test/drlg_l3_test.cpp

```diff
@@ -0,0 +1,124 @@
+#include <fmt/format.h>
+#include <gtest/gtest.h>
+
+#include "drlg_l3.h"
+#include "engine/load_file.hpp"
+#include "gendung.h"
+#include "quests.h"
+#include "utils/paths.h"
+
+using namespace devilution;
+
+namespace {
+
+void TestCreateL3Dungeon(int level, uint32_t seed, lvl_entry entry)
+{
+	paths::SetPrefPath(paths::BasePath());
+
+	std::string dunPath;
+
+	if (level >= 9 && level <= 12) {
+		dunPath = fmt::format("test/fixtures/diablo/{}-{}.dun", level, seed);
+		pMegaTiles = std::make_unique<MegaTile[]>(206);
+		leveltype = DTYPE_CAVES;
+	} else if (level >= 17 && level <= 20) {
+		dunPath = fmt::format("test/fixtures/hellfire/{}-{}.dun", level, seed);
+		pMegaTiles = std::make_unique<MegaTile[]>(166);
+		leveltype = DTYPE_NEST;
+	}
+
+	currlevel = level;
+	CreateL3Dungeon(seed, entry);
+
+	auto dunData = LoadFileInMem<uint16_t>(dunPath.c_str());
+	ASSERT_NE(dunData, nullptr) << "Unable to load test fixture " << dunPath;
+	ASSERT_EQ(Size(DMAXX, DMAXY), Size(dunData[0], dunData[1]));
+
+	const uint16_t *tileLayer = &dunData[2];
+
+	for (int y = 0; y < DMAXY; y++) {
+		for (int x = 0; x < DMAXX; x++) {
+			auto tileId = static_cast<uint8_t>(SDL_SwapLE16(*tileLayer));
+			tileLayer++;
+			ASSERT_EQ(dungeon[x][y], tileId) << "Tiles don't match at " << x << "x" << y;
+		}
+	}
+
+	const uint16_t *transparentLayer = &dunData[2 + DMAXX * DMAXY * 13];
+
+	for (int y = 16; y < 16 + DMAXY * 2; y++) {
+		for (int x = 16; x < 16 + DMAXX * 2; x++) {
+			auto sectorId = static_cast<uint8_t>(SDL_SwapLE16(*transparentLayer));
+			transparentLayer++;
+			ASSERT_EQ(dTransVal[x][y], sectorId) << "Room/region indexes don't match at " << x << "x" << y;
+		}
+	}

```

Comment:
This is similar to to `TestCreateL5Dungeon` and `TestCreateL2Dungeon` (+`SetPrefPath`).
Do we want to put this in a function and call it in all tests?

Reply:
Yes, I promised to do it as the next PR :)